### PR TITLE
User daily rewards

### PR DESCRIPTION
**Pull request by robertg93**

### CODE REVIEW

File: sdk/src/utils.ts [hunk 1]
```diff
@@ -1213,6 +1213,42 @@ export const positionsRewardAPY = (params: ApyPositionRewardsParams): number =>
...
+  try {
+    const { singleTickLiquidity } = calculateTokensAndLiquidity(
+      ticksCurrentSnapshot,
+      currentTickIndex
+    )
+
+    const dailyRewards = rewardInTokens / duration
+
+    const liquidityRatio =
+      userLiquidity.v.mul(LIQUIDITY_DENOMINATOR).div(singleTickLiquidity).toNumber() /
```
Comment (wojciech-cichocki) [inline sdk/src/utils.ts:1241]: `allActiveLiquidity` instead of `singleTickLiquidity`

**Review by wojciech-cichocki — CHANGES_REQUESTED** [review]

File: tests/math.spec.ts [hunk 2]
```diff
@@ -2037,6 +2041,56 @@ describe('Math', () => {
...
+        currentTickIndex: usdcUsdhPoolSnapshot.currentTickIndex,
+        rewardInTokens: 10000,
+        userLiquidity: { v: singleTickLiquidity.divn(5) },
+        duration: 10,
+        lowerTickIndex: tickLower,
+        upperTickIndex: tickUpper
+      }
+
+      const apy = calculateUserDailyRewards(paramsApy)
+      //10000 tokens / 10 days * 20% of liquidity = 2000 tokens
```
Comment (wojciech-cichocki) [inline tests/math.spec.ts:2066]: `200` in comment. This var should name `dailyRewards` or  something like that

File: tests/math.spec.ts [hunk 3]
```diff
@@ -2037,6 +2041,56 @@ describe('Math', () => {
...
+        ticksCurrentSnapshot: current,
+        currentTickIndex: usdcUsdhPoolSnapshot.currentTickIndex,
+        rewardInTokens: 10000,
+        userLiquidity: { v: singleTickLiquidity.divn(5) },
+        duration: 10,
+        lowerTickIndex: tickLower,
+        upperTickIndex: tickUpper
+      }
+
+      const apy = calculateUserDailyRewards(paramsApy)
```
Comment (wojciech-cichocki) [inline tests/math.spec.ts:2089]: same `dailyRewards` instead of `apy`

**Review by wojciech-cichocki — APPROVED** [review]

**Review by wojciech-cichocki — CHANGES_REQUESTED** [review]
Small changes

File: sdk/src/utils.ts [hunk 4]
```diff
@@ -994,7 +995,11 @@ export const calculateTokensRange = (
     throw new Error(Errors.TickNotFound)
   }
 
-  const tokens = getTokensInRange(tickArrayCurrent, tickLower, tickUpper)
+  const tokensPrevious = getTokensInRange(tickArrayPrevious, tickLower, tickUpper)
+  const tokensCurrent = getTokensInRange(tickArrayCurrent, tickLower, tickUpper)
+
+  // geometric mean of tokensPrevious and tokensCurrent
+  const tokens = sqrt(tokensPrevious.mul(tokensCurrent))
```
[the remark this answers is not in the document]
Comment (robertg93) [reply sdk/src/utils.ts:1002]: before this section there is a check if one of the arrays is empty, therefore in this case both are 'not empty'

File: tests/math.spec.ts (hunk 2 above)
Comment (robertg93) [reply tests/math.spec.ts:2066]: done

File: tests/math.spec.ts (hunk 3 above)
Comment (robertg93) [reply tests/math.spec.ts:2089]: done

**Review by wojciech-cichocki — APPROVED** [review]

**Pull request merged**